prow/README.md

```diff
@@ -4,206 +4,383 @@ Metal3 Prow dashboard: <https://prow.apps.test.metal3.io>
+   # so is the application credentials!
+   openstack application credential create prow-capo-cloud-controller
+   ```
 
-1. Create a [bucket](https://cloud.google.com/storage/docs/creating-buckets)
-1. Create a
-   [service account](https://cloud.google.com/iam/docs/creating-managing-service-accounts)
-1. On the created bucket, grant `storage.objects.create` access for the service
-   account and `storage.objectViewer` permission to `allUser`.
+   The output of this command should give you an application credential ID and secret.
+   Store them in a safe place.
+   In the rest of this document they will be referred to as `${APP_CRED_ID}` and
+   `${APP_CRED_SECRET}`.
 
-## Setup
+1. Set up S3 object storage.
 
-Prow was set up by following these instructions:
-<https://github.com/kubernetes/test-infra/blob/master/prow/getting_started_deploy.md>
+   ```bash
+   openstack --os-interface public ec2 credentials create
+   ```
 
-1. Create DNS record(s) to point to your cluster.
-1. Prow components are running on `default` namespace, while `test-pods`
-   namespace is used by pods running the actual CI jobs.
-1. [ghProxy](https://github.com/kubernetes/test-infra/tree/master/ghproxy) is a
-   reverse proxy, and is meant to reduce API token usage. We need persistent
-   volume for ghProxy, and for that purpose we rely on `hostPath` type of
-   `PersistentVolume` in our cluster due to the limitation in the underlying
-   infrastructure. Ideally, you should use a dynamic storage provisioner.
+   Store the access key and secret key in a safe place.
+   They will be needed also for deploying prow.
+   They will be referred to as `${S3_ACCESS_KEY}` and `${S3_SECRET_KEY}`.
 
-   Create a /mnt/prow directory to store the persistent volume data:
+1. Generate HMAC token.
 
    ```shell
-   sudo mkdir /mnt/prow
+   openssl rand -hex 20
    ```
 
-   Create the PersistentVolume:
+   It will be referred to as `${HMAC_TOKEN}`.
 
-   ```shell
-   kubectl apply -f persistent_volume.yaml
-   ```
+## GitHub configuration
 
-1. Grant a user `cluster-admin` role in all namespaces to create cluster
-   resources.
+1. Create bot accounts.
+   The bot accounts are normal accounts on GitHub.
+   Both of them have Gmail accounts connected to the GitHub accounts.
 
-   ```shell
-   $ kubectl create clusterrolebinding cluster-admin-binding-"${USER}" \
-      --clusterrole=cluster-admin --user="${USER}"
-   ```
+1. Create a personal access token for the GitHub bot account.
+   This should be done from the [metal3-io-bot](https://github.com/metal3-io-bot)
+   GitHub bot account.
+   You can follow this [link](https://github.com/settings/tokens) to create the
+   token. When generating the token, make sure you have only the following scopes
+   checked in.
 
-1. Create HMAC token for webhook validation and generate a secret out of it.
-   Don't forget to add this token on GitHub webhook configuration.
+   - `repo` scope for full control of private repositories
```

Comment:
Instead of private, maybe say `all` repositories. I know its Github wording, but I found it annoying when its just implicitly including public ones.